### ISSUE TITLE
ReadOnly child types cause NotSupportedException when calling BeginEdit

### ISSUE THREAD

**Issue by j055** [MarimerLLC/csla#1525]
**Describe the bug**
Cannot use ReadOnly child types and NLevel undo. When calling ```BeginEdit``` a ```NotSupportedException``` is thrown. This seems to be related to migrating from using the ```BinaryFormatter``` to the ```MobileFormatter```. 

If I change the ```ReadOnlyListBase``` and ```ReadOnlyBase``` to their Business equivalents then it works.

**Version and Platform**
CSLA version: 5.1
OS: Windows, Linux, iOS, Android, etc.
Platform: ASP.NET Core

**Code that Fails**
```Csla.Core.BusinessBase.BeginEdit()```

**Stack Trace or Exception Detail**
```
System.NotSupportedException
  HResult=0x80131515
  Message=Serialization of type object is not supported.
  Source=Csla
  StackTrace:
   at Csla.Serialization.Mobile.CslaBinaryWriter.Write(Object target, BinaryWriter writer)
   at Csla.Serialization.Mobile.CslaBinaryWriter.Write(Stream serializationStream, List`1 objectData)
   at Csla.Serialization.Mobile.MobileFormatter.Serialize(Stream serializationStream, Object graph)
   at Csla.Core.FieldManager.FieldDataManager.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.Core.UndoableBase.CopyState(Int32 parentEditLevel)
   at Csla.Core.UndoableBase.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.BusinessListBase`2.CopyState(Int32 parentEditLevel)
   at Csla.BusinessListBase`2.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.Core.FieldManager.FieldDataManager.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.Core.UndoableBase.CopyState(Int32 parentEditLevel)
   at Csla.Core.UndoableBase.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.BusinessListBase`2.CopyState(Int32 parentEditLevel)
   at Csla.BusinessListBase`2.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.Core.FieldManager.FieldDataManager.Csla.Core.IUndoableObject.CopyState(Int32 parentEditLevel, Boolean parentBindingEdit)
   at Csla.Core.UndoableBase.CopyState(Int32 parentEditLevel)
   at Csla.Core.BusinessBase.BeginEdit()
   at Business.Core.BusinessBase`1.BeginEdit(Int32 level) in D:\source\repos\Website\Business\Core\BusinessBase.cs:line 20

```

**Comment by rockfordlhotka** [comment]
With the BinaryFormatter any urecognized type (from an undo perspective) is simply serialized as a binary blob.

I'm not entirely sure why this is different for MobileFormatter and undo, as anything that implements `IMobileObject` should be serializable. 

Needs research.

**Comment by j055** [comment]
Is a fix for this going to get into 5.2? Do you need a repo or is it a straight forward fix?

**Comment by rockfordlhotka** [comment]
@j055 I actually haven't researched this at all - I took your word that there's an issue and haven't created a repo of my own yet.
